DiamondLightSource/SynchWeb
# LIMS-1074: Allow accessing zipped files

**JIRA ticket**: [LIMS-1074](https://jira.diamond.ac.uk/browse/LIMS-1074)

**Summary**:

To save on disk space, many files created by auto processing pipelines have been zipped (*.gz), which breaks the links to them in Synchweb, as well as a number of image files and map / model viewers. These should be fixed server-side.

**Changes**:
- Use gzopen and gzread to temporarily unzip the processing files before serving them to the front end
- Add already zipped files to archive.zip file served by "Archive" button (they download as unzipped files zipped together)
- Fix the map / model viewer shell script to unzip the relevant files first
- Fix 2 silly spelling mistakes

**To test**:
- View a pre-2022 data collection with autoprocessing, eg /dc/visit/mx23694-64/id/6669874
- For fast_dp
  - Check the 'Archive' button downloads a full zip file
  - Check the 'Plots' button produces graphs
  - Check the 'Radiation Damage' button produces a graph
  - View the 'Logs & Files' modal, and try viewing files of type log and graph and downloading files of type log, graph and result
  - Repeat tests for different autoprocessing pipelines
- For downstream processing
  - Check the 'Archive' and 'Logs & Files' buttons as above
  - Check the 'Map / Model Viewer' button loads a model
  - Check thumbnails of models load for Autosharp, Autobuild and Crank2
- Check no error messages appear on display or in dev console
